newscout_web/api/v1/views.py

```diff
@@ -174,23 +174,27 @@ def post(self, request, format=None):
         """
         Save new category to database
         """
-        serializer = CategorySerializer(data=request.data, many=True)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(create_response(serializer.data))
-        return Response(create_error_response(serializer.errors), status=400)
+        if request.user.is_authenticated:
+            serializer = CategorySerializer(data=request.data, many=True)
+            if serializer.is_valid():
+                serializer.save()
+                return Response(create_response(serializer.data))
+            return Response(create_error_response(serializer.errors), status=400)
+        raise Http404
 
     def put(self, request, format=None):
         """
         update category in database
         """
-        _id = request.data.get("id")
-        obj = Category.objects.get(id=_id)
-        serializer = CategorySerializer(obj, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(create_response(serializer.data))
-        return Response(create_error_response(serializer.errors), status=400)
+        if request.user.is_authenticated:
+            _id = request.data.get("id")
+            obj = Category.objects.get(id=_id)
```

Comment:
rename `obj` to `category`

Reply:
fixed in commit id: 9a96e142828f5174f24905b18c33c5daa3b07dec